presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java

```diff
@@ -147,7 +147,21 @@ protected HiveTableLayoutHandle(
         this.layoutString = requireNonNull(layoutString, "layoutString is null");
         this.requestedColumns = requireNonNull(requestedColumns, "requestedColumns is null");
         this.partialAggregationsPushedDown = partialAggregationsPushedDown;
-        this.appendRowNumberEnabled = appendRowNumberEnabled;
+        // if $row_id is requested, we need the row numbers whether they're explicitly requested or not
+        // TODO is it possible someone needs the row_numbers and row IDs, and thus we need a block for each?
+        // in which case we might need to know whether the row numbers were explicitly requested, and thus
+        // need a separate variable for appending row IDs
+        // TODO verify that $row_id can appear in any of these column lists
```

Comment:
can you do this todo and then have the code do only what is necessary?

Reply:
done